packages/@aws-cdk/aws-appmesh/lib/tls-validation.ts

```diff
@@ -125,11 +164,110 @@ class TlsValidationFileTrust extends TlsValidationTrust {
           certificateChain: this.certificateChain,
         },
       },
+      virtualNodeListenerTlsValidationTrust: {
+        file: {
+          certificateChain: this.certificateChain,
+        },
+      },
       virtualGatewayClientTlsValidationTrust: {
         file: {
           certificateChain: this.certificateChain,
         },
       },
+      virtualGatewayListenerTlsValidationTrust: {
+        file: {
+          certificateChain: this.certificateChain,
+        },
+      },
     };
   }
 }
+
+class TlsValidationSdsTrust extends TlsValidationTrust {
+  /**
+   * The name of the secret for Envoy to fetch from a specific endpoint through the Secrets Discovery Protocol.
+   */
+  readonly secretName: string;
+
+  constructor (props: TlsValidationSdsTrustOptions) {
+    super();
+    this.secretName = props.secretName;
+  }
+
+  public bind(_scope: Construct): TlsValidationTrustConfig {
+    return {
+      virtualNodeClientTlsValidationTrust: {
+        sds: {
+          secretName: this.secretName,
+        },
+      },
+      virtualNodeListenerTlsValidationTrust: {
+        sds: {
+          secretName: this.secretName,
+        },
+      },
+      virtualGatewayClientTlsValidationTrust: {
+        sds: {
+          secretName: this.secretName,
+        },
+      },
+      virtualGatewayListenerTlsValidationTrust: {
+        sds: {
+          secretName: this.secretName,
+        },
+      },
+    };
+  }
+}
+
+/**
+ * All Properties for Subject Alternative Names Matcher for both Client Policy and Listener.
+ */
+export interface SubjectiveAlternativeNamesMatcherConfig {
+
+  /**
+   * VirtualNode CFN configuration for subject alternative names secured by the certificate.
+   */
+  readonly virtualNodeSans: CfnVirtualGateway.SubjectAlternativeNameMatchersProperty;
```

Comment:
nit: you're not exposing it to the customer anywhere so this just a style thing but I like `virtualNodeSubjectAlternativeNames` instead of `virtualNodeSans`. It's could be confusing that some variable names have it spelled out and some have it abbreviated.

Reply:
Well noted. I switched the abbreviated to full name to match the style.